api/server/server.go

```diff
@@ -1640,8 +1640,9 @@ func allocateDaemonPort(addr string) error {
 		return fmt.Errorf("failed to lookup %s address in host specification", host)
 	}
 
+	pa := portallocator.Get()
```

Comment:
I would just follow the conventions of the flag package. Don't worry about the cost of checking `sync.Once`. It will not be a bottleneck.